### ISSUE TITLE
On the Calculation of Loss Function of Discriminator

### ISSUE THREAD

**Issue by zhouhao-learning** [eriklindernoren/Keras-GAN#122]
Hello, I'm glad to see your tutorial. When I study GAN, I see the following part of the code. I don't understand why `d_loss` calculates like this. I don't understand the meaning of this calculation. Please guide me. Thank you.

![image](https://user-images.githubusercontent.com/24581773/53158528-cbe10d80-35fe-11e9-8d64-c83b83293b18.png)

**Comment by niunno** [comment]
If I've understood your question:
the discriminator tries to distinguish real images from fake ones, which are created by the generator; therefore, the discriminator is trained to output "valid" when the input is imgs (real images) and "fake" when the input is gen_imgs (generated images).

**Comment by zhouhao-learning** [comment]
In fact, I do not understand the following calculation method:

```
d_loss = 0.5 * np.add(d_loss_real, d_loss_fake)
```
what is the significance of this calculation?

**Comment by niunno** [comment]
I think it is a mean. If "valid" and "fake" have the same size:
d_loss = (n * d_loss_real + n * d_loss_fake) / ( 2 * n) = n * (d_loss_real + d_loss_fake) / (2 * n) = 0.5 * (d_loss_real + d_loss_fake)

**Comment by zhouhao-learning** [comment]
Okay, thank you
